src/D2L.CodeStyle.Analyzers/UnsafeStatics/UnsafeStaticsAnalyzer.cs

```diff
@@ -169,42 +169,32 @@ string fieldOrPropName
 				return;
 			}
 
-			// try to use the concrete type if we have it (via a constructor)
 			var flags = MutabilityInspectionFlags.Default;
-			var constructedType = GetConstructedType( context, exp );
-			if( constructedType != null ) {
-				type = constructedType;
-				flags |= MutabilityInspectionFlags.AllowUnsealed;
-			}
 
-			if( m_immutabilityInspector.IsTypeMutable( type, flags ) ) {
-				var diagnostic = CreateDiagnostic( location, fieldOrPropName, type.GetFullTypeNameWithGenericArguments() );
-				context.ReportDiagnostic( diagnostic );
-			}
-		}
+			// Always prefer the type from the initializer if it exists because
+			// it may be more specific.
+			if( exp != null ) {
+				var typeInfo = context.SemanticModel.GetTypeInfo( exp );
 
-		private ITypeSymbol GetConstructedType(
-			SyntaxNodeAnalysisContext context,
-			ExpressionSyntax exp
-		) {
+				if( typeInfo.Type is IErrorTypeSymbol ) {
+					return;

```

Comment:
This can happen when you have a syntax error.

I'll do a follow-up PR that makes our tests always assert there aren't any types of non-our-analyzer errors because right now it's possible to create test cases that pass when they shouldn't if you make a typo in the test string (this confused me for awhile...)